daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1886,15 +1912,16 @@ private[lf] object Speedy {
   /** Continuation marking the exit of a closure. This is only used during
     * profiling.
     */
-  private[speedy] final case class KLeaveClosure(label: Profile.Label) extends Kont {
-    override def execute[Q](machine: Machine[Q], v: SValue): Control.Value = {
+  private[speedy] final case class KLeaveClosure[Q](machine: Machine[Q], label: Profile.Label)
+      extends Kont[Q] {
+    override def execute(v: SValue): Control.Value = {
       machine.profile.addCloseEvent(label)
       Control.Value(v)
     }
   }
 
-  private[speedy] final case object KPreventException extends Kont {
-    override def execute[Q](machine: Machine[Q], v: SValue): Control.Value = {
+  private[speedy] final case class KPreventException[Q]() extends Kont[Q] {
```

Comment:
Are polymorphic objects disallowed?

Reply:
yes polymorphic 'object' cannot be created, this is why I change the `case object` in to a `case class`.